### PR TITLE
Registration progress

### PR DESCRIPTION
**Pull request by aaronamoso**
Fixes: #15 

## Description
Please include a summary of the changes and the related issue. List any dependencies that are required for this change.

## Feature Link:

---

## Description:
Finished reg form
---

## Screenshots:
Attach screenshots or recordings of your changes here.
Screenshot 2024-12-05 213509
Screenshot 2024-12-05 213500

---

## 📢 Notes for Reviewers
Add any additional information or context for the reviewers.

I just couldnt figure out how to make it responsive. Also, I have not finished input validation. Only used "required" attribute on the TS.

### CODE REVIEW

File: src/App.tsx
Comment (AKalyanPavan) [inline src/App.tsx:1]: @aaronamoso, Attempting to make the page responsive is an awesome effort.Please pull the changes from the main branch and adjust the code accordingly.

File: src/App.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ export default function App() {
           <BrowserRouter>
             <Routes>
               <Route path="/" element={<BaseLayout />}>
+                <Route path="/register" element={<Register />} />
                 <Route path="/login" element={<Login />} />
```
Comment (AKalyanPavan) [inline src/App.tsx:31]: Path for registration-page is better when it's presented outside of the <BaseLayout /> element, just like the sign-in page present in the main. 


Screenshot 2024-12-07 at 3 40 54 AM

File: src/pages/register/register.css
Comment (AKalyanPavan) [inline src/pages/register/register.css:1]: This css structure looks good.

File: src/pages/register/register.tsx
Comment (AKalyanPavan) [inline src/pages/register/register.tsx:1]: in this file, we can make the code little bit optimized by following DRY principle. We can create an object of input fields and can render them using a .map function, which can reduce the number of lines of code. We can use the same DRY principle while rendering <option> elements too!

**Comment by aaronamoso** [comment]
I have pulled the code for the changes you have implemented. Though, I am unsure which areas you need me to work on besides optimizing register.tsx by following DRY principle.

**Comment by AKalyanPavan** [comment]
> I have pulled the code for the changes you have implemented. Though, I am unsure which areas you need me to work on besides optimizing register.tsx by following DRY principle.

Apart from DRY principle, we have to change the path for registration-page in App.tsx. But I am sure you've already understood this by hitting a like. Right now as I checked, registration_progress branch is still behind, from the main branch. Please merge with main branch and change path in App.tsx. Apart from that everything looks 👍.

Thank You.

**Comment by aaronamoso** [comment]
I did push the optimized code but I am not entirely sure how to merge my code to the main. I may have fixed AKalyanPavan's issue but please let me know! This is my first time working with a team especially using git.

**Comment by HongyuLiuLance** [comment]
> I did push the optimized code but I am not entirely sure how to merge my code to the main. I may have fixed AKalyanPavan's issue but please let me know! This is my first time working with a team especially using git.

Before you try to push your code to the main branch, you need to pull the latest main branch code to merge into your branch and resolve conflicts. When you try to merge code into the main branch, it needs to pass at least two approvals. Please update the code and resolve the conflict issue first, then submit the pr again.

**Comment by AKalyanPavan** [comment]
> I did push the optimized code but I am not entirely sure how to merge my code to the main. I may have fixed AKalyanPavan's issue but please let me know! This is my first time working with a team especially using git.

Hi @aaronamoso , 

I can understand that this is your first time with a team using git. I am sorry for that. If you have a difficulty in merging, create a new branch from main and add your changes in that branch. Then we don't have to worry about merging and conflicts, as everything present in main will also be present in your new branch too. Raise a new pull request and we can proceed further.
@HongyuLiuLance, please keep a note about new pull request, to smoothen this process. 

And I have checked your branch. 
I still find <Register /> inside <BaseLayout /> element.

Screenshot 2024-12-09 at 2 48 28 PM

 If that is the case the output looks like this.
 
Screenshot 2024-12-09 at 2 49 56 PM

But we want the registration of a user should take place even before the user should sign-in. If we change our code as shown in the following image,

Screenshot 2024-12-09 at 2 51 30 PM

The output will be like this.

Screenshot 2024-12-09 at 2 52 47 PM

With a header and footer of course. 

By the way, I checked your register.tsx page. You nailed it. It's an awesome work. 🎊 .

**Comment by aaronamoso** [comment]
Thank you for your detailed response. I will get to the moment I get home.
Thank you!

**Comment by aaronamoso** [comment]
Screenshot 2024-12-09 184630
Okay, I have move the component before signin!

**Comment by aaronamoso** [comment]
I have also created a new PR called newBranchREGISTRATIONForm

**Review by tianchengc — APPROVED** [review]

**Pull request merged**